### PR TITLE
rustc_typeck exists no more

### PR DESCRIPTION
**Pull request by tshepang**
part of #1491

cc @lcnr

### CODE REVIEW

**Review by JohnTitor — APPROVED** [review]
LGTM!

File: src/type-checking.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # Type checking
 
-The [`rustc_typeck`][typeck] crate contains the source for "type collection"
+The [`hir_analysis`] crate contains the source for "type collection"
 and "type checking", as well as a few other bits of related functionality. (It
 draws heavily on the [type inference] and [trait solving].)
```
Comment (lcnr) [inline src/type-checking.md:5]: this isn't completely right as we've moved type check to `rustc_hir_typeck`.

maybe something like
```suggestion
The [`hir_analysis`] crate contains the source for "type collection" as well as a bunch of related functionality. Checking the bodies of functions is implemented in the [`hir_typeck`] crate. (It draws heavily on the [type inference] and [trait solving].)
```
```

Comment (tshepang) [reply src/type-checking.md:5]: which of these 2 crates "draws heavily..."

**Comment by JohnTitor** [comment]
Should we also correct the description of https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir_analysis/index.html? It has some description for type checking.

File: src/type-checking.md (hunk 1 above)
Comment (lcnr) [reply src/type-checking.md:5]: both do :grin:

**Comment by tshepang** [comment]
is CI stuck

**Pull request merged**